### PR TITLE
Prevent error in creating relative path accross driver letters in Windows. Fixes #320

### LINKED ISSUE #320: Filters don't work across Windows drive boundaries
I have _project in D drive_, the _compiler tool chain in C drive_.
When I try generating code coverage, it tries to generate coverage report for the files from compiler tool chain also, Even after explicitly mentioning the --root=<ProjectRoot> from D drive.

How can I inform GCOVR tool that changing the drive for reports is crime:-).

**This actually causing some python utility error in my machine**
ValueError: path is on mount 'c:', start on mount 'D:'

_Originally posted by @the-code-cult in https://github.com/gcovr/gcovr/issues/293#issuecomment-507248288_

### PR DESCRIPTION
**Pull request by AbdManian**

### CODE REVIEW

**Comment by latk** [comment]
Thank you!

* Could you add a short entry to the changelog explaining this bugfix?
* If you want to, you can add a line to the AUTHORS file.

**Comment by AbdManian** [comment]
> 
> 
> Thank you!
> 
>     * Could you add a short entry to the changelog explaining this bugfix?
> 
>     * If you want to, you can add a line to the AUTHORS file.

I am new to git and github. I've updated changelog in my fork. Is it correct? Should I submit new pull request?

**Comment by latk** [comment]
> I am new to git and github. I've updated changelog in my fork. Is it correct? Should I submit new pull request?

No it's good. All new commits on your `master` branch will also be part of this pull request. You can see in the timeline of this PR that your commit on the changelog was picked up.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -59,6 +59,7 @@ Internal changes:
  - Remove Appveyor CI and upload coverage report from Windows and Ubuntu from the GitHub actions. (:issue:`455`)
  - Add check if commit is mentioned in the CHANGELOG.rst. (:issue:`457`)
  - Move flake8 config to setup.cfg and add black code formatter. (:issue:`444`)
+ - Fix filter/exclude relative path issue in Windows. (:issue:`320`)
```
Comment (latk) [inline CHANGELOG.rst:62]: Our automated tests look for mentions of this PR. Once that is fixed, the unit tests should run.

```suggestion
 - Fix filter/exclude relative path issue in Windows. (:issue:`320`, :issue:`479`) 
```

File: gcovr/utils.py [hunk 2]
```diff
@@ -217,6 +217,15 @@ def __init__(self, root, pattern):
...
     def match(self, path):
         abspath = os.path.realpath(path)
+
+        # On Windows, a relative path can never cross drive boundaries.
+        # If so, the relative filter cannot match.
+        if sys.platform == 'win32':
+            path_drive, _ = os.path.splitdrive(abspath)
+            root_drive, _ = os.path.splitdrive(os.path.realpath(self.root))
+            if path_drive != root_drive:
+                return None
```
Comment (Spacetown) [inline gcovr/utils.py:227]: Can't be checked without using a `subst` command. Missing coverage is OK.

**Review by Spacetown — APPROVED** [review]

**Pull request merged**